src/sentry/sentry_apps/utils/errors.py

```diff
@@ -1,49 +1,47 @@
+from collections.abc import Mapping, Sequence
 from enum import Enum
 
+from sentry.utils import json
+
 
 class SentryAppErrorType(Enum):
     CLIENT = "client"
     INTEGRATOR = "integrator"
     SENTRY = "sentry"
 
 
-# Represents a user/client error that occured during a Sentry App process
-class SentryAppError(Exception):
-    error_type = SentryAppErrorType.CLIENT
-    status_code = 400
+class SentryAppBaseError(Exception):
+    error_type: SentryAppErrorType
+    status_code: int
 
     def __init__(
         self,
         error: Exception | None = None,
         status_code: int | None = None,
     ) -> None:
-        if status_code:
-            self.status_code = status_code
+        self.status_code = status_code or self.status_code
+        self.error = error
+
+    def __str__(self) -> str:
+        if isinstance((Mapping, Sequence), self.error.args[0]):
+            return json.dumps(self.error.args[0])
```

Comment:
what is `error.args[0]`? is there ever more than 1 argument? is there a way to get the specific field you're looking for with `getattr`?

Reply:
When creating making Exceptions, the `BaseException` class puts all args into a tuple which is the` self.args` property. So the inner part of `Exception({"cool": "cat"}, "cooool")` would be ({"cool": "cat"}, "cooool"). So yeah it's possible to have multiple args to an exception but I think it's highly unlikely since people can just add to a dict if they have multiple things (and multi arg exceptions are really rare in the codebase).